command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
```

Comment:
Yeah I typed it in a text box so it's not 100%. I'm sure you could get around the issue you mentioned :). I'd definitely prefer to be explicit using channel code to coordinate rather than handling some of the code inside a timer, then stopping the timer elsewhere. Right now the coordinationg code is sprinkled between `CombinedOutput`, `timerWithTimeout`, and `runCommand`, and it's very difficult to follow what's happening. With my suggestion you'd have all the coordination in one place and the code flows linearly.